daal4py/sklearn/model_selection/split.py

```diff
@@ -0,0 +1,149 @@
+import daal4py as d4p
+import numpy as np
+
+try:
+    import mkl_random
+    mkl_random_is_imported = True
+except (ImportError, ModuleNotFoundError):
+    mkl_random_is_imported = False
+
+try:
+    import pandas as pd
+    pandas_is_imported = True
+except ImportError:
+    pandas_is_imported = False
+
+
+def get_dtypes(data):
+    if hasattr(data, 'dtype'):
+        return [data.dtype]
+    elif hasattr(data, 'dtypes'):
+        return list(data.dtypes)
+    elif hasattr(data, 'values'):
+        return [data.values.dtype]
+    else:
+        return None
+
+
+def _daal_train_test_split(*arrays, **options):
+    n_arrays = len(arrays)
+    if n_arrays == 0:
+        raise ValueError("At least one array required as input")
+    test_size = options.pop('test_size', None)
+    train_size = options.pop('train_size', None)
+    random_state = options.pop('random_state', None)
+    stratify = options.pop('stratify', None)
+    shuffle = options.pop('shuffle', True)
+    rng = options.pop('rng', 'MT19937')
+
+    available_rngs = ['MT19937', 'SFMT19937', 'MT2203', 'R250', 'WH', 'MCG31',
+                      'MCG59', 'MRG32K3A', 'PHILOX4X32X10', 'NONDETERM']
+    if rng not in available_rngs:
+        raise ValueError(
+            "Wrong random numbers generator is chosen. "
+            "Available generators: %s" % str(available_rngs)[1:-1])
+
+    if options:
+        raise TypeError("Invalid parameters passed: %s" % str(options))
+
+    arrays = indexable(*arrays)
+
+    n_samples = _num_samples(arrays[0])
+    n_train, n_test = _validate_shuffle_split(n_samples, test_size, train_size,
+                                              default_test_size=0.25)
+    if shuffle is False:
+        if stratify is not None:
+            raise ValueError(
+                "Stratified train/test split is not implemented for "
+                "shuffle=False")
+
+        train = np.arange(n_train)
+        test = np.arange(n_train, n_train + n_test)
+    else:
+        if stratify is not None:
+            cv = StratifiedShuffleSplit(test_size=n_test,
+                                        train_size=n_train,
+                                        random_state=random_state)
+            train, test = next(cv.split(X=arrays[0], y=stratify))
+        else:
+            if mkl_random_is_imported and (isinstance(random_state, int) or random_state is None):
+                random_state = mkl_random.RandomState(random_state, rng)
+                indexes = random_state.permutation(n_train + n_test)
+                train, test = indexes[:n_train], indexes[n_train:]
+            else:
+                cv = ShuffleSplit(test_size=n_test,
+                                  train_size=n_train,
+                                  random_state=random_state)
+                train, test = next(cv.split(X=arrays[0], y=stratify))
+
+    res = []
+    for arr in arrays:
+        fallback = False
+        if not isinstance(arr, np.ndarray):

```

Comment:
Maybe we can make a check for `numpy` and `pandas` input in a common place? I think it would be good for @owerbat  to have a check too. What do you think?